src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -268,5 +270,45 @@ private static string GetExpectedRedirectUri(string clientId)
         {
             return $"ms-appx-web://microsoft.aad.brokerplugin/{clientId}";
         }
+
+        /// <summary>
+        /// Converts to MSAL Account Id or Null
+        /// </summary>
+        /// <param name="wamAccount"></param>
+        /// <param name="clientID"></param>
+        public static IAccount ConvertToMsalAccount(
+                NativeInterop.Account wamAccount, 
+                string clientID)
+        {
+            IAccount runtimeAccount;
+
+            try
+            {
+                string homeAccountId = GetHomeAccountId(wamAccount);
+
+                runtimeAccount = new Account(
+                    wamAccount.AccountId,
+                    wamAccount.UserName,
+                    wamAccount.Environment,
+                    new Dictionary<string, string>() { { clientID, wamAccount.AccountId } });
+
+            }
+            catch (Exception ex)
```

Comment:
the only thing that can fail here is the `GetHomeAccountId`. You can throw better errors (MsalClientExceptions) for thibgs like "no username", "no account id" etc.